### ISSUE TITLE
Learning deep representations by mutual information estimation and maximization

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#20]
## [Learning deep representations by mutual information estimation and maximization](https://arxiv.org/abs/1808.06670)
*In this work, we perform unsupervised learning of representations by maximizing mutual information between an input and the output of a deep neural network encoder. Importantly, we show that structure matters: incorporating knowledge about locality of the input to the objective can greatly influence a representation's suitability for downstream tasks. We further control characteristics of the representation by matching to a prior distribution adversarially. Our method, which we call Deep InfoMax (DIM), outperforms a number of popular unsupervised learning methods and competes with fully-supervised learning on several classification tasks. DIM opens new avenues for unsupervised learning of representations and is an important step towards flexible formulations of representation-learning objectives for specific end-goals.*

### Cite this paper
```bibtex
@inproceedings{hjelm2018learning,
  title={Learning deep representations by mutual information estimation and maximization},
  author={R Devon Hjelm and Alex Fedorov and Samuel Lavoie-Marchildon and Karan Grewal and Phil Bachman and Adam Trischler and Yoshua Bengio},
  booktitle={International Conference on Learning Representations},
  year={2019},
  url={https://openreview.net/forum?id=Bklr3j0cKX},
}
```

**Comment by juanmc2005** [comment]
[A presentation of this article](https://www.youtube.com/watch?v=o1HIkn8LEsw)

**Comment by juanmc2005** [comment]
### Intro
This article introduces a new technique for self-supervised representation learning called Deep InfoMax, which learns by maximizing mutual information between representations of different levels in the neural network hierarchy. Even though this technique is task-agnostic, their experiments are conducted on computer vision tasks.

### Architecture
The architecture consists of:
- An encoder of convolutions transforms an input image x into an MxM patch matrix
- A fully connected layer transforms patches into a representation y
- A discriminator of real and fake examples
- A discriminator for adversarially matching the representation distribution to a prior

### Objective
The training objective of Deep InfoMax is quite complex, as it tries to do 3 things simultaneously:

1. Maximizing global mutual information
2. Maximizing local mutual information
3. Matching representations to a prior distribution via adversarial training

#### 1. Global InfoMax
GIM is the most simple approach, they sample n images from the training data and compute patches and representations. Positive pairs are made of these patches and representations.
Then they sample n different images from the training data and they compute the patch matrix, these patches are paired with the initial representations to form the negative pairs.
They maximize mutual information between the matrix of patches and y using a discriminator of real (positive) and fake (negative) examples.

![image](https://user-images.githubusercontent.com/5760780/76948869-fbe50800-6907-11ea-9d8d-6e1ea9bc2e0c.png)

#### 2. Local InfoMax
This one is essentially equivalent to GIM, but pairs are formed using individual patches from the matrix and not the whole matrix, so mutual information is maximized between local patches and y.

![image](https://user-images.githubusercontent.com/5760780/76948914-08696080-6908-11ea-98f3-6b003311da47.png)

#### 3. Matching to a prior
They match to a prior distribution via adversarial training, by training a discriminator to tell apart examples drawn from the prior and from the encoder.

### Results
They have found that using different objectives leads to better results, both in terms of global and local features, and in terms of mutual information estimation. In general, as global features are not encouraged to rely on relevant abstract features of the input and may learn pixel-level details, LIM tends to obtain better performance and to better cluster similar examples:

![image](https://user-images.githubusercontent.com/5760780/76948829-ea036500-6907-11ea-9af5-c5d6255293f2.png)

### Conclusion
This is a rather general explanation of the basic method of described in the paper. It is interesting that they use a similar technique as contrastive or triplet loss, but by optimizing mutual information instead of a distance. Another interesting aspect is the adversarial training that allows for powerful unsupervised contrastive training. It might be worth diving deeper into this.

Another thing to note is that pixel-level detail and noise is still being captured by these models. This remains one of the most difficult challenges of unsupervised representation learning: how to distinguish the signal from the noise when no ground truth is given.
